### PR TITLE
FEATURE: better `ansi -e` error 

### LINKED ISSUE #8704: improve the errors in `ansi -e`
### Related problem

as i did with `error make` in #8511, i think we could have slightly better error message for the `ANSI` format record given to `ansi -e` :yum: 

for now we have things like
```bash
>_ ansi -e
Error: nu::shell::missing_parameter

  × Missing parameter: code.
   ╭─[entry #19:1:1]
 1 │ ansi -e
   · ──┬─
   ·   ╰── missing parameter: code
   ╰────
```

```bash
>_ ansi -e {}
```

```bash
>_ ansi -e {foo: "foo"}
Error: nu::shell::incompatible_parameters

  × Incompatible parameters.
   ╭─[entry #21:1:1]
 1 │ ansi -e {foo: "foo"}
   ·         ──────┬─────
   ·               ╰── problem with key: foo
   ╰────
```

### Describe the solution you'd like

i think it would be a bit better to explain
- what should be expected when one gives nothing to `ansi -e`
- what is the "problem" with the "foo" key

and to add a complete example to the `help` of `ansi` :+1: 

### Describe alternatives you've considered

_No response_

### Additional context and details

_No response_

### PR DESCRIPTION
**Pull request by amtoine**
Should close #8704.

# Description
this PR
- makes the error thrown by things like `ansi -e {invalid: "invalid"}` more explicit
- makes the `ansi -e` example more explicit about valid / invalid keys

# User-Facing Changes
the error
```bash
> ansi -e {invalid: "invalid"}
Error: nu::shell::incompatible_parameters

  × Incompatible parameters.
   ╭─[entry #1:1:1]
 1 │ ansi -e {invalid: "invalid"}
   ·         ──────────┬─────────
   ·                   ╰── unknown ANSI format key: expected one of ['fg', 'bg', 'attr'], found 'invalid'
   ╰────
```

the new `ansi -e` example
```bash
  Use structured escape codes
  > let bold_blue_on_red = {  # `fg`, `bg`, `attr` are the acceptable keys, all other keys are considered invalid and will throw errors.
        fg: '#0000ff'
        bg: '#ff0000'
        attr: b
    }
    $"(ansi -e $bold_blue_on_red)Hello Nu World(ansi reset)"
  Hello Nu World
```

# Tests + Formatting
- :green_circle: `toolkit fmt`
- :green_circle: `toolkit clippy`
- :black_circle: `toolkit test`
- :black_circle: `toolkit test stdlib`

# After Submitting
```
$nothing
```

### CODE REVIEW

File: crates/nu-command/src/platform/ansi/ansi_.rs [hunk 1]
```diff
@@ -608,7 +608,12 @@ Format: #
             },
             Example {
                 description: "Use ansi to color text with a style (blue on red in bold)",
-                example: r#"$"(ansi -e { fg: '#0000ff' bg: '#ff0000' attr: b })Hello Nu World(ansi reset)""#,
+                example: r#"let format = {  # none of the following keys are mandatory, but any other key will be considered invalid
```
Comment (fdncred) [inline crates/nu-command/src/platform/ansi/ansi_.rs:611]: I'm not sure that text is correct. Those are the only acceptable keys, but they're not all mandatory. e.g. I have this in my config and I think it works fine.
```
    leading_trailing_space_bg: { attr: n } # no fg, no bg, attr none, this effectively turns this off
```

Comment (amtoine) [reply crates/nu-command/src/platform/ansi/ansi_.rs:611]: i like the text you propose, it's clearer :ok_hand:

Comment (amtoine) [reply crates/nu-command/src/platform/ansi/ansi_.rs]: done in 66843de068786805794a3555c6fc0150a3cc2946 with the addition that any invalid keys will give an error :yum:

**Comment by fdncred** [comment]
apparently there are conflicts to resolve, or i'd land this.

**Comment by amtoine** [comment]
> apparently there are conflicts to resolve, or i'd land this.
>
should be fixed in e926354ef3bf10a90b01c906b0c5b00c2de6689f :yum: 

if the tests pass, i think we're good with these new errors :muscle:

**Pull request merged**